app/uk/gov/hmrc/trackingconsentfrontend/assets/test/ui/renderBanner.spec.ts

```diff
@@ -42,22 +50,34 @@ describe('renderBanner', () => {
   }
 
   it('should render a banner', () => {
-    renderBanner(userPreference)
+    renderBanner(userPreference, preferenceCommunicator)
 
     expect(queryByText(document.body, tellUsYouAcceptAllMatcher)).toBeTruthy()
   })
 
   it('should call preference manager when user accepts', () => {
-    renderBanner(userPreference)
+    renderBanner(userPreference, preferenceCommunicator)
     assume(userAcceptsAll).not.toHaveBeenCalled()
 
     clickAcceptAll()
 
     expect(userAcceptsAll).toHaveBeenCalledWith()
   })
 
+  it('should call communicator after setting preferences', () => {
```

Comment:
Just re-iterating, might be worth considering moving this responsibility to the userPreference object. If we add more interfaces in future, the UI code is going to get polluted with concerns around managing the communication of preferences to other systems which I don't think is it's responsibility.

Reply:
That's fair, I'll try moving the communicator function back into the `userPreferences` and see whether that fits better.